src/release_notes/model/base_chapters.py

```diff
@@ -0,0 +1,63 @@
+from abc import ABC, abstractmethod
+from release_notes.model.chapter import Chapter
+from release_notes.model.record import Record
+
+
+class BaseChapters(ABC):
+    """
+    An abstract base class representing the base chapters.
+    """
+
+    def __init__(self, sort_ascending: bool = True, print_empty_chapters: bool = True):
+        """
+        Constructs all the necessary attributes for the BaseChapters object.
+
+        :param sort_ascending: A boolean indicating whether to sort the chapters in ascending order.
+        :param print_empty_chapters: A boolean indicating whether to print empty chapters.
+        """
+        self.sort_ascending = sort_ascending
+        self.print_empty_chapters = print_empty_chapters
+        self.chapters: dict[str, Chapter] = {}
+
+    def add_row(self, chapter_key: str, number: int, row: str):
```

Comment:
I like how you are using typing everywhere. Why not to enforce the type check then? :) In CI or so - look for `mypy`

there even is GH action for this - I never used it but looks good on the first glance: https://github.com/jpetrucciani/mypy-check

Reply:
Add as new task in Epic.